walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -11,13 +11,72 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
+ 
+public final class FindMeetingQuery { 
 
-public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> requiredBusyTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> allBusyTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> viableTimeSlots;
+    int duration = (int)request.getDuration();
+
+    for(Event e : events) {
+      // Ensure event includes requireds
+      if (eventContains(e.getAttendees(), request.getAttendees())) {
+        requiredBusyTimes.add(e.getWhen());  // Build list of only requireds
+        allBusyTimes.add(e.getWhen()); // Add required list to overall list
+      }  
+      if(eventContains(e.getAttendees(), request.getOptionalAttendees()))
+        allBusyTimes.add(e.getWhen()); // Add optionals to overall list
+    }
+    
+    // Generate available times including optionals
+    generatePotentialTimes(allBusyTimes, duration);
+ 	if(viableTimeSlots.size() == 0) {
```

Comment:
nit: space between "if" and "(".
Please fix the ones in the for loop/if statements below as well.

Reply:
Also the indentation of the line is off. Should align with line 37.